src/dbus/clientprivate.cpp

```diff
@@ -87,112 +89,40 @@ Ngf::ClientPrivate::~ClientPrivate()
-                         this, SLOT(serviceRegistered(const QString&)));
         QObject::connect(m_serviceWatcher, SIGNAL(serviceUnregistered(const QString&)),
                          this, SLOT(serviceUnregistered(const QString&)));
 
-        // When connecting to bus for the first time, check whether NGFD exists in the bus.
-        // After we know the initial state it is enough to follow NameOwnerChanged.
-        QDBusMessage msg = QDBusMessage::createMethodCall(QStringLiteral("org.freedesktop.DBus"),
-                                                          QStringLiteral("/org/freedesktop/DBus"),
-                                                          QStringLiteral("org.freedesktop.DBus"),
-                                                          QStringLiteral("GetNameOwner"));
-        QList<QVariant> args;
-        args << NgfDestination;
-        msg.setArguments(args);
-
-        QDBusMessage reply = QDBusConnection::systemBus().call(msg);
-
-        if (reply.type() == QDBusMessage::ErrorMessage) {
-            changeConnected(false);
-            changeAvailable(false);
-            return false;
-        } else {
-            changeAvailable(true);
-        }
-    }
-
-    if (!m_available) {
-        changeConnected(false);
-        return false;
-    }
-
-    QDBusInterface *iface;
-    iface = new QDBusInterface(NgfDestination, NgfPath, NgfInterface,
-                               QDBusConnection::systemBus(), this);
-
-    // Match NGFD signals to get statuses for ongoing events
-    if (iface) {
-        if (iface->isValid()) {
-            iface->connection().connect(QString(), NgfPath, NgfInterface, SignalStatus,
-                                        this, SLOT(setEventState(quint32,quint32)));
-            m_iface = iface;
-            changeConnected(true);
-        } else {
-            iface->deleteLater();
-        }
+        QDBusConnection::systemBus().connect(QString(), NgfPath, NgfInterface, SignalStatus,
+                                             this, SLOT(setEventState(quint32,quint32)));
     }
 
+    // connected doesn't mean much really, mostly just backward compatibility
+    changeConnected(true);
     return m_connected;
 }
 
 void Ngf::ClientPrivate::disconnect()
 {
-    m_connectionWanted = false;
-
-    if (m_iface) {
-        m_iface->deleteLater();
-        m_iface = 0;
-    }
-
     changeConnected(false);
 }
 
-void Ngf::ClientPrivate::serviceRegistered(const QString &service)
-{
-    Q_UNUSED(service);
-
-    changeAvailable(true);
-
-    // NGFD has reappeared in system bus, let's reconnect
-    if (m_connectionWanted)
-        connect();
-}
-
 void Ngf::ClientPrivate::serviceUnregistered(const QString &service)
 {
     Q_UNUSED(service);
 
-    // NGFD has died for some reason
-    changeAvailable(false);
     // All currently active events are invalid, so clear event list
     removeAllEvents();
```

Comment:
Noting that ngfd is not dbus autostart service + now events are sent without tracking service availablity -> afaict events added during service down time are left in the books, or? Just thinking whether tracking also WatchForRegistration just to do removeAllEvents() also on that trigger would help with that or not?

Reply:
> afaict events added during service down time are left in the books, or?

Those cases will eventually receive method call error response, and the handling does removeEvent(event);

It's an option to take into account the known state of the service and avoid keeping temporary events that are known to be failing, but don't know if it's worth it.